src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -404,12 +404,12 @@ private IRoleClaimStore<TRole> GetClaimStore()
         /// <param name="user"></param>
         /// <param name="methodName"></param>
         /// <returns></returns>
-        protected async Task<IdentityResult> LogResultAsync(IdentityResult result,
+        protected Task<IdentityResult> LogResultAsync(IdentityResult result,
             TRole role, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
         {
-            result.Log(Logger, Resources.FormatLoggingResultMessageForRole(methodName, await GetRoleIdAsync(role)));
+            result.Log(Logger, () => { return Resources.FormatLoggingResultMessageForRole(methodName, GetRoleIdAsync(role).Result); });
```

Comment:
Hmmm. Sprinkling code that can potentially block everywhere isn't good. Let's chat about this more. We should probably hold of making this change until there is a version of `Log()` that takes an async lambda.